mne_bids_pipeline/steps/sensor/_05_decoding_csp.py

```diff
@@ -253,6 +254,21 @@ def _fmt_contrast(cond1, cond2, fmin, fmax, freq_range_name, tmin=None, tmax=Non
         freq_decoding_table.loc[idx, "mean_crossval_score"] = cv_scores.mean()
         freq_decoding_table.at[idx, "scores"] = cv_scores
 
+        # PATTERNS
+        csp.fit_transform(X, y)
+        sensor_pattern_csp = csp.patterns_
+
+        # COEFS
+        clf.fit(X, y)
+        weights_csp = mne.decoding.get_coef(clf, "patterns_", inverse_transform=True)
```

Comment:
Typically this is all I use. Why also take the `csp.patterns_`? `csp.fit(X, y)` results anywhere else so it's a bit weird to me to add them here. It seems like we should just use these `weights_csp`. And maybe we should call them `clf_patterns_` because they're really the patterns inverse transformed from the CSP all the way back through the other steps (e.g., PCA, sensor scaling)?

Reply:
that is exactly where I am lost in the terminology.. 

my approach was to save them in this step to pick them up in a separate source projection step later

Reply:
I wanted to save both, because in @britta-wstnr's code they are combined during beamforming:

```
stc_csp = beamform_components(weights_csp, sensor_pattern_csp, spat_filter,
                                fwd, multipliers=multiplier)
```

Reply:
The idea here is to combine the coefficients and the CSP components. The (Haufe et al.) patterns are not easily beamform-able as they do not represent classical sink-source patterns anymore but refer to the decoding classes. Using the CSP components is a work-around for that.